app/src/main/java/org/fossasia/openevent/app/common/di/component/BarcodeComponent.java

```diff
@@ -0,0 +1,24 @@
+package org.fossasia.openevent.app.common.di.component;
+
+import org.fossasia.openevent.app.common.di.module.BarcodeModule;
+import org.fossasia.openevent.app.common.di.module.DataModule;
+import org.fossasia.openevent.app.common.di.module.NetworkModule;
+import org.fossasia.openevent.app.common.di.module.PresenterModule;
+import org.fossasia.openevent.app.qrscan.ScanQRActivity;
+
+import javax.inject.Singleton;
+
+import dagger.Component;
+
+@Singleton
+@Component(modules = {
+    DataModule.class,
+    NetworkModule.class,
+    PresenterModule.class,
+    BarcodeModule.class
+})
+public interface BarcodeComponent {
```

Comment:
It does not need all these modules. Basically, you inject 2 components in ScanQRActivity, one the common app component and other one this